src/chat/flagsReducers.js

```diff
@@ -0,0 +1,58 @@
+import {
```

Comment:
Why not just keep it as part of `state.messages` and immutably update the message whose flag has changed?